src/pages/token/rewards/UnbondingPanels.tsx

```diff
@@ -0,0 +1,119 @@
+import { shallowEqual } from 'react-redux';
+import styled from 'styled-components';
+import { formatUnits } from 'viem';
+
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStakingValidator } from '@/hooks/useStakingValidator';
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useTokenConfigs } from '@/hooks/useTokenConfigs';
+
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AssetIcon } from '@/components/AssetIcon';
+import { Output, OutputType } from '@/components/Output';
+import { Panel } from '@/components/Panel';
+import { ValidatorFaviconIcon } from '@/components/ValidatorName';
+
+import { getUnbondingDelegations } from '@/state/accountSelectors';
+import { useAppSelector } from '@/state/appTypes';
+
+export const UnbondingPanels = () => {
+  const stringGetter = useStringGetter();
+  const unbondingDelegations = useAppSelector(getUnbondingDelegations, shallowEqual);
+  const { unbondingValidators } = useStakingValidator() ?? {};
+  const { chainTokenLabel, chainTokenDecimals } = useTokenConfigs();
+
+  if (!unbondingDelegations?.length) {
```

Comment:
hmm curious about how you feel about this approach vs passing in `unbondingDelegations` to `UnbondingPanels` (and having the caller just not call `UnbondingPanels` if there are no `unbondingDelegations`). I sort of prefer the latter because I think it's unexpected for a component to return `null`

Reply:
I don't have a strong opinion, but I recently changed a few components to return null sometimes. I think it makes sense for whatever business logic determines whether a component is shown or not to be in the component so you don't have to essentially duplicate it and leak the details into the parent.

Reply:
don't feel to strongly either. I think I mainly did it this way so that the view can encapsulate its own business logic.

Reply:
Yep exactly